core/lib/sync/task.ts

```diff
@@ -182,3 +183,44 @@ export function initializeSyncTaskStorage () {
     driver: localForage[config.localForage.defaultDrivers['syncTasks']]
   }))
 }
+
+export function registerSyncTaskProcessor () {
+  const mutex = {}
+  Vue.prototype.$bus.$on('sync/PROCESS_QUEUE', data => {
+    if (typeof navigator !== 'undefined' && navigator.onLine) {
+      // event.data.config - configuration, endpoints etc
+      const syncTaskCollection = Vue.prototype.$db.syncTaskCollection
+      const currentUserToken = rootStore.getters['user/getUserToken']
+      const currentCartToken = rootStore.getters['cart/getCartToken']
+
+      const fetchQueue = []
+      Logger.debug('Current User token = ' + currentUserToken)()
+      Logger.debug('Current Cart token = ' + currentCartToken)()
+      syncTaskCollection.iterate((task, id, iterationNumber) => {
+        if (task && !task.transmited && !mutex[id]) { // not sent to the server yet
+          mutex[id] = true // mark this task as being processed
+          fetchQueue.push(() => {
+            return execute(task, currentUserToken, currentCartToken).then(executedTask => {
+              syncTaskCollection.removeItem(id) // remove successfully executed task from the queue
+              mutex[id] = false
+            }).catch(err => {
+              mutex[id] = false
+              Logger.error(err)()
+            })
+          })
+        }
+      }, (err, result) => {
+        if (err) Logger.error(err)()
+        Logger.debug('Iteration has completed')()
+        // execute them serially
+        serial(fetchQueue)
```

Comment:
method should be async and we could just do
```js
for (const fetchRequest of fetchQueue) await fetchRequest
Logger.debug('Processing sync tasks queue has finished')()
```

Reply:
Done - however with the serial helper kept in place